### ISSUE TITLE
Replace uses of Go package "io/ioutil" when the minimum supported Go is 1.16

### ISSUE THREAD

**Issue by krader1961** [elves/elvish#1350]
I noticed that the Go "io/ioutil" package has been deprecated: https://golang.org/doc/go1.16#ioutil. Elvish makes extensive use of that package. When Go 1.17 is released we should update this project to use the alternatives documented by that link.

**Comment by krader1961** [comment]
@xiaq just merged commit 9076934395344d308180dd36ba289285dd7af779 that addresses this issue so closing it.

**Comment by krader1961** [comment]
Reopening because I noticed there are still a handful of uses of package `io/ioutil`:

```
> grep ioutil **.go
pkg/edit/complete/generators.go:        "io/ioutil"
pkg/edit/complete/generators.go:        infos, err := ioutil.ReadDir(dirToRead)
pkg/fsutil/claim.go:    "io/ioutil"
pkg/fsutil/claim.go:    files, err := ioutil.ReadDir(dir)
pkg/fsutil/search.go:   "io/ioutil"
pkg/fsutil/search.go:           infos, _ := ioutil.ReadDir(dir)
pkg/glob/glob.go:       "io/ioutil"
pkg/glob/glob.go:// readDir is just like ioutil.ReadDir except that it treats an argument of ""
pkg/glob/glob.go:       return ioutil.ReadDir(dir)
```
